zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -226,32 +416,80 @@ impl<F: Field> ExecutionGadget<F> for BeginTxGadget<F> {
                 //   - Write TxAccessListAccount
                 //   - Write Account Balance
                 //   - Write Account Balance
-                //   - Read Account CodeHash (only if tx is not create)
-                //   - Write CallContext Depth
-                //   - Write CallContext CallerAddress
-                //   - Write CallContext CalleeAddress
-                //   - Write CallContext CallDataOffset
-                //   - Write CallContext CallDataLength
-                //   - Write CallContext Value
-                //   - Write CallContext IsStatic
-                //   - Write CallContext LastCalleeId
-                //   - Write CallContext LastCalleeReturnDataOffset
-                //   - Write CallContext LastCalleeReturnDataLength
-                //   - Write CallContext IsRoot
-                //   - Write CallContext IsCreate
-                //   - Write CallContext CodeHash
-                rw_counter: Delta(22.expr() + (1.expr() - tx_is_create.expr())),
+                //   - Read Account CodeHash
+                rw_counter: Delta(10.expr()),
                 call_id: To(call_id.expr()),
-                is_root: To(true.expr()),
-                is_create: To(tx_is_create.expr()),
-                code_hash: To(phase2_code_hash.expr()),
-                gas_left: To(gas_left),
-                reversible_write_counter: To(2.expr()),
-                log_id: To(0.expr()),
-                ..StepStateTransition::new_context()
+                ..StepStateTransition::any()
             });
         });
 
+        // 4. Call to account with non-empty code.
+        cb.condition(
+            and::expr([
+                not::expr(tx_is_create.expr()),
+                not::expr(is_empty_code_hash.expr()),
+            ]),
+            |cb| {
+                // Setup first call's context.
+                for (field_tag, value) in [
+                    (CallContextFieldTag::Depth, 1.expr()),
+                    (CallContextFieldTag::CallerAddress, tx_caller_address.expr()),
+                    (CallContextFieldTag::CalleeAddress, tx_callee_address.expr()),
+                    (CallContextFieldTag::CallDataOffset, 0.expr()),
+                    (
+                        CallContextFieldTag::CallDataLength,
+                        tx_call_data_length.expr(),
+                    ),
+                    (CallContextFieldTag::Value, tx_value.expr()),
+                    (CallContextFieldTag::IsStatic, 0.expr()),
```

Comment:
How do we know that `IsStatic = false`? We aren't in a create Tx but we should still be able to be in a non-static call.

Reply:
Nevermind. The root call always needs to be non-static right? @ed255 do you happen to know?

Reply:
Yes, I forgot to respond to this. My understanding was that a contract creation transaction (`is_create * is_root == true`) MUST be non-static.